WordPress/WordPressApi/WordPressComOAuthClient.swift

```diff
@@ -0,0 +1,189 @@
+import AFNetworking
+
+@objc public enum WordPressComOAuthError: Int {
+    case Unknown
+    case InvalidClient
+    case UnsupportedGrantType
+    case InvalidRequest
+    case NeedsMultifactorCode
+}
+
+/**
+ `WordPressComOAuthClient` encapsulates the pattern of authenticating against WordPress.com OAuth2 service.
+
+ Right now it requires a special client id and secret, so this probably won't work for you
+
+ @see https://developer.wordpress.com/docs/oauth2/
+ */
+
+public class WordPressComOAuthClient: NSObject {
+
+    public static let WordPressComOAuthErrorDomain = "WordPressComOAuthError"
+    public static let WordPressComOAuthKeychainServiceName = "public-api.wordpress.com"
```

Comment:
I know this was already there, but just leaving a note that I don't think it belongs here. The OAuth client doesn't care about the Keychain. `WPAccount` is the only class using this constant and it should be the one defining it.